packages/block-library/src/navigation/block-colors-selector.js

```diff
@@ -58,27 +58,46 @@ const renderToggleComponent = ( { value } ) => ( { onToggle, isOpen } ) => {
 				label={ __( 'Open Colors Selector' ) }
 				onClick={ onToggle }
 				onKeyDown={ openOnArrowDown }
-				icon={ <ColorSelectorIcon color={ value } /> }
+				icon={ <ColorSelectorIcon color={ textColor } backgroundColor={ backgroundColor } /> }
 			/>
 		</ToolbarGroup>
 	);
 };
 
-const renderContent = ( { value, onChange = noop } ) => ( () => {
+const renderContent = ( {
+	textColor,
+	onTextColorChange = noop,
```

Comment:
I wonder whether you need to consider preserving the old API for this component by aliasing `onTextColorChange` to `onChange`. This might help preserve backwards compatibility across other components.

Isn't there some kind of deprecation utility in Gutenberg which you can call programmatically to silently warn developers they are consuming an old API?